docs/references/tanstack-start/create-clerk-handler.mdx

```diff
@@ -0,0 +1,50 @@
+---
+title: '`createClerkHandler()`'
+description: The createClerkHandler() is used to configure the Clerk for your TanStack Start application.
+---
+
+The `createClerkHandler()` function configures Clerk to handle authentication state for TanStack routes, allowing you to easily access user session information within your app.
+
+## Usage
+
+### Basic usage
+
+The following example demonstrates how to use `createClerkHandler()` to configure Clerk in your TanStack Start application.
+
+```tsx {{ filename: 'app/ssr.tsx' }}
+import { createStartHandler, defaultStreamHandler } from '@tanstack/start/server'
+import { getRouterManifest } from '@tanstack/start/router-manifest'
+import { createRouter } from './router'
+import { createClerkHandler } from '@clerk/tanstack-start/server'
+
+export default createClerkHandler(
+  createStartHandler({
+    createRouter,
+    getRouterManifest,
+  }),
+)(defaultStreamHandler)
+```
+
+### Advanced usage
+
+The following example demonstrates how to configure `createClerkHandler` with dynamic keys.
```

Comment:
We have [a document about dynamic keys for Next.js](https://clerk.com/docs/references/nextjs/clerk-middleware#dynamic-keys) - did we want to link that here?
```suggestion
The following example demonstrates how to configure `createClerkHandler()` with [dynamic keys](/docs/references/nextjs/clerk-middleware#dynamic-keys){{ target: '_blank' }}.
```

Reply:
We might need to change the name, it's similar but not quite in the same context as we don't have middleware in TanStack Start, I will try to update this section!